tsdb/head_read.go

```diff
@@ -599,14 +624,36 @@ type boundedChunk struct {
 }
 
 func (b boundedChunk) Bytes() []byte {
```

Comment:
The `Bytes()` logic for `boundedChunk` and `mergedOOOChunks` are almost the same (apart from `boundedChunk` not panicking when `xc.Appender()`  returns an error - I think it should be panicking in this cases actually). Could we extract the bytes logic into a function that both `boundedChunk` and `mergedOOOChunks` use?